File: src/arduinobot_cpp_examples/src/simple_subscriber.cpp
#include<rclcpp/rclcpp.hpp>
#include<std_msgs/msg/string.hpp>

class SimpleSubscriber : public rclcpp::Node{
    public:
        SimpleSubscriber(): Node("simple_subscriber"){
            sub_ = create_subscription<std_msgs::msg::String>(
                "chatter", 10, std::bind(&SimpleSubscriber::callback, this, std::placeholders::_1));

        }
        void callback(const std_msgs::msg::String::SharedPtr msg) const {
            RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
        }
    private:
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;    
};

int main(int argc, char * argv[]){
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SimpleSubscriber>());
    rclcpp::shutdown();
    return 0;
}
// }
// This code is a simple ROS 2 subscriber that listens to a topic named "chatter"
// and prints the received messages to the console. It uses the std_msgs/msg/String message type.

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleSubscriber(Node):
    def __init__(self):
        super().__init__('simple_subscriber')
        self.sub_ = self.create_subscription(String, "chatter", self.msg_callback, 10)
        self.get_logger().info("Simple Subscriber Node has been started.")
    def msg_callback(self, msg):
        self.get_logger().info(f"Received message: {msg.data}")


def main(args=None):
    rclpy.init(args=args)
    simple_subscriber= SimpleSubscriber()
    try:
        rclpy.spin(simple_subscriber)
    except KeyboardInterrupt:
        pass
    finally:
        simple_subscriber.destroy_node()
        rclpy.shutdown()
if __name__ == '__main__':
    main()
    
    
# This code defines a simple ROS 2 subscriber node that listens to the "chatter" topic        